sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+
+ * Now the ``PLT[0]`` entry pushes the ``ip0`` and ``lr`` registers
+   onto the stack.  The dynamic linker will later use the stacked
+   value of ``ip0`` to compute a relocation index by subtracting the
+   base of the ``GOT`` The index will be used to select an entry from
+   the table of relocation entries addressed by the ``DT_JMPREL``
+   dynamic section entry. The designated relocation entry must have
+   type ``R_AARCH64_JUMP_SLOT``, and its offset will specify the
+   global offset table entry used by the branch instruction in
+   ``PLT[N]``, and also a symbol table index that references the
+   appropriate symbol, i.e. ``name1`` in this example.
+
+ * The ``PLT[0]`` entry then loads the adress of the third GOT entry
+   (``.got.plt[2]``) into ``ip0``, and the loads and jumps to the
+   address in the third table entry, which transfers control to the
+   dynamic linker.
+
+ * The value in ``ip0`` is used by the dynamic linker's entrypoint to
+   load the second GOT entry (``.got.plt[2]``) using ``[ip0,#-8]``,
+   which gives it one word of private information with which to
+   identify the calling module.
+
+ * The dynamic linker unwinds the stack, looks at the designated
+   relocation entry, finds the symbol's value, stores the actual
+   address of ``name1`` in ``.got.plt[N]`` , then transfers control to
+   the desired destination.  Subsequent executions of ``PLT[N]`` will
+   transfer control directly to ``name1``, without calling the dynamic
+   linker a second time. That is the ``BR`` instruction in ``PLT[N]``
+   will transfer to ``name1``, instead of ``PLT[0]``.
+
+The ``LD_BIND_NOW`` environment variable can change the dynamic linking
+behavior. If its value is non-null, the dynamic linker evaluates all
+procedure linkage table entries before transferring control to the
+program. That is, the dynamic linker processes relocation entries of
+type ``R_AARCH64_JUMP_SLOT`` during process initialization. Otherwise, the
+dynamic linker evaluates procedure linkage table entries lazily,
+delaying symbol resolution and relocation until the first call to a
+table entry.
+
+The ``PLT[0]`` entry calls the dynamic linker's entrypoint, stored in
+``.got.plt[2]``, with the following calling convention:
+
++---------------+-------------------------+
+| Location      |    Contents             |
++===============+=========================+
+| ``ip0``       |   ``&.got.plt[2]``      |
++---------------+-------------------------+
+| ``ip1``       | DL Resolver Entry point |
++---------------+-------------------------+
+| ``[sp, #0]``  |  ``&.got.plt[N]``       |
++---------------+-------------------------+
+| ``[sp, #8]``  | ``lr``                  |
++---------------+-------------------------+
+
+Sample PLT sequences
+^^^^^^^^^^^^^^^^^^^^
+
+The following section shows some examples of PLT sequences suitable
+for the small code model. There is an assumption that the ``.plt.got``
+section is within +- 4 GiB from the ``.plt`` section.
```

Comment:
Thanks for the suggestion, I think it can be used without making the text too complicated.

Reply:
.plt.got => .got.plt

Reply:
Ack. Thanks.

Reply:
The 4GB limit applies to **all** code models, so it's best not to mention `suitable for the small code model` because the sequences apply to all code models.

Reply:
OK. Will remove that part.